src/cdk/table/array-data-source.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DataSource} from './data-source';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+/**
+ * Array-based data source that can be used to connect to a CollectionViewer. Will trigger an
+ * update to the CollectionViewer whenever a new data array is set.
+ */
+export class ArrayDataSource<T> implements DataSource<T> {
+  /** Stream that emits when a new data array is set on the data source. */
+  private _data: BehaviorSubject<T[]>;
+  set data(data: T[]) { this._data.next(data); }
+  get data() { return this._data.value; }
+
+  constructor(initialData: T[] = []) {
+    this._data = new BehaviorSubject<T[]>(initialData);
+  }
+
+  /**
+   * Triggers an update to the CollectionViewer. Should be used when elements are added, removed,
+   * or moved in the data array and should be reflected in the CollectionViewer.
+   */
+  update() { this._data.next(this.data); }
```

Comment:
I think both of those suggestions have merit, and I do like the idea of making it clear that data is updated through one path only. Keep in mind that changes made to the data itself (excluding adding/removing rows) will still automatically get picked up by change detection and that may confused users when they need to call `update`.

E.g. If you change the `name` property on the first object in the array, then that change will automatically be reflected in the table through standard view change detection without having to call `update()`. Check out the unit test to see that I can change `a_1` to `changeda_1` without calling `update()`.

@jelbourn's prefers we go the path of excluding the update function altogether and teaching people that they always need to set the data property to see things update.

Not sure what the best approach is, each has the confusing aspect that the table changes some times and other times needs a trigger. In the meantime, I'm going to add in sorting and pagination =P